### PR TITLE
[Fixes #282] Implement a CLI tool to guide in the creation of the .env file

### LINKED ISSUE #282: Implement a CLI tool to guide in the creation of the .env file
The .env file has several repeated configurations, some with slightly different representation (domain with or without scheme, port, etc.).
Editing the .env file is error prone.

A cli tool to guide in the creation of the .env file would simplify the process and make it more robust.

### Features
- it should be a pure Python command, without any dependency outside Python builtins
- it will accept arguments and parameters either as: 
   - CLI arguments
   - JSON file
- it will use create the .env file starting from the .env.sample file
- it will overwrite the .env file on each run

### Parameters
 - required:
   - `hostname`, e.g. `master.demo.geonode.org`
 - optional:
   - `https` (boolean), default value is `False`
   - `email` - Admin email (this is required if https is set to True, since a valid email is required by Letsencrypt certbot)
   - `geonodepwd` - GeoNode admin password (required inside the .env)
   - `geoserverpwd` - Geoserver admin password (required inside the .env)
   - `pgpwd` - PostgreSQL password (required inside the .env)
   - `dbpwd` - GeoNode DB user password (required inside the .env)
   - `geodbpwd` - Geodatabase user password (required inside the .env)
   - `clientid` - Oauth2 client id (required inside the .env)
   - `clientsecret` - Oauth2 client secret (required inside the .env)

In case the optional (but required) parameters are missing, the tool will populate them with randomly generated strings.

The parameters can also be fed with a JSON file (a -f, --file option can be used to tell the CLI to use the file instead of arguments)

```json
{
	"hostname": ...,
	"https": ...,
	"email": ...,
	"geonodepwd": ...,
	"geoserverpwd": ...,
	"pgpwd": ...,
	"dbpwd": ...,
	"geodbpwd": ...,
	"clientid": ...,
	"clientsecret": ...,
} 
```

### .env substitutions
The .env.sample will have the following placeholders that will be replaced by the cli tool.

- POSTGRES_PASSWORD: `pgpwd`
- GEONODE_DATABASE_PASSWORD: `dbpwd`
- GEONODE_GEODATABASE_PASSWORD: `geodbpwd`
- DATABASE_URL: postgis://{GEONODE_DATABASE}:{`dbpwd`}@db:5432/{GEONODE_DATABASE}
- GEODATABASE_URL: postgis://{GEONODE_GEODATABASE}:{`geodbpwd`}@db:5432/{GEONODE_GEODATABASE}
- ALLOWED_HOSTS: "['django', '{`hostname`}', '*']"
- GEONODE_LB_HOST_IP: `hostname`
- HTTP_HOST: empty if `https` is True, otherwise `hostname`
- HTTPS_HOST: empty if `https` is False, otherwise `hostname`
- LETSENCRYPT_MODE: 'disabled' if https is False, otherwise 'production'
- GEOSERVER_WEB_UI_LOCATION: http(s)`hostname` (https if https is set to True)
- GEOSERVER_PUBLIC_LOCATION: http(s)`hostname` (https if https is set to True)
- GEOSERVER_ADMIN_PASSWORD: `geoserverpwd`
- ADMIN_PASSWORD: `geonodepwd`
- ADMIN_EMAIL: `email`
- OAUTH2_CLIENT_ID: `clientid`
- OAUTH2_CLIENT_SECRET: `clientsecret`

### PR DESCRIPTION
**Pull request by mattiagiupponi**
ref #282 

**NOTE**: if both JSON file and CLI args are passed, the CLI args will overwrite the JSON value
Example usage:
```bash
python create-envfile.py localhost -f /opt/core/geonode-project/file.json \
    --https \
    --email random@email.com \
    --geonodepwd gn_password \
    --geoserverpwd gs_password \
    --pgpwd pg_password \
    --dbpwd db_password \
    --geodbpwd _db_password \
    --clientid 12345 \
    --clientsecret abc123 
```
In attach the output of the above command (I had to convert it as TXT otherwise Github doesn't accept it. the script will create the .env file)
[env.txt](https://github.com/GeoNode/geonode-project/files/8652099/env.txt)

### CODE REVIEW

File: create-envfile.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    with open(f"{dir_path}/.env", "w+") as output_env:
+        output_env.write(_sample_file)
+    logger.info(f".env file created: {dir_path}/.env")
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        prog="ENV file builder",
+        description="Tool for generate environment file automatically. The information can be passed or via CLI or via JSON file ( --file /path/env.json)",
+        usage="python build_env.py localhost -f /path/to/json/file.json",
```
Comment (afabiani) [inline create-envfile.py]: Change the help string to `python create-envfile ...`

File: create-envfile.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
+import argparse
```
Comment (afabiani) [inline create-envfile.py:20]: Add headers to the new files

File: create-envfile.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+_strong_chars = shuffle(string.ascii_letters + \
+    string.digits + \
+    string.punctuation.replace("\"", '').replace("'", '').replace("`", ''))
+
+
+def generate_env_file(parser):
+    ################################################################
+    #######                     Required                    ########
+    ################################################################
+    parser.add_argument("hostname", help="host name")
```
Comment (afabiani) [inline create-envfile.py]: I would add a default value here `localhost`

File: README.md [hunk 4]
```diff
@@ -46,7 +41,55 @@ To setup your project using a local python virtual environment, follow these ins
...
+    - *dbpwd*: GeoNode DB user password (required inside the .env)
+    - *geodbpwd*: Geodatabase user password (required inside the .env)
+    - *clientid*: Oauth2 client id (required inside the .env)
+    - *clientsecret*: Oauth2 client secret (required inside the .env)
+    - *secret key*: Django secret key (required inside the .env)
+    - *sample_file*: absolute path to a env_sample file used to create the env_file. If not provided, the one inside the GeoNode project is used.
+    - *file*: absolute path to a json file that contains all the above configuration
+
+     **NOTE:**
+    - if the same configuration is passed in the json file and as an argument, the CLI one will overwrite the one in the JSON file
```
Comment (afabiani) [inline README.md:66]: Can we put an example of `file.json` contents please?

**Review by afabiani — CHANGES_REQUESTED** [review]

File: create-envfile.py (hunk 3 above)
Comment (mattiagiupponi) [reply create-envfile.py]: Was requested to be a required parameter, providing a default makes it an optional param

File: README.md (hunk 4 above)
Comment (mattiagiupponi) [reply README.md:66]: I'll add it to the README, meanwhile if you want to see it, is present also in the issue

File: create-envfile.py (hunk 3 above)
Comment (giohappy) [reply create-envfile.py]: it was requested to be required, but I agree with @afabiani.

Comment (mattiagiupponi) [reply create-envfile.py]: Done 👍

**Review by afabiani — APPROVED** [review]

**Pull request merged**